zbus/tests/e2e.rs

```diff
@@ -473,6 +473,30 @@ async fn my_iface_test(conn: Connection, event: Event) -> zbus::Result<u32> {
     }
     drop(stream);
 
+    let root_introspect_proxy = zbus::fdo::IntrospectableProxy::builder(&conn)
+        .destination("org.freedesktop.MyService")?
+        .path("/")?
+        .build()
+        .await?;
+    debug!("Created: {:?}", root_introspect_proxy);
+
+    let root_xml = root_introspect_proxy.introspect().await?;
+    let root_node = zbus_xml::Node::from_reader(root_xml.as_bytes())
+        .map_err(|e| Error::Failure(e.to_string()))?;
```

Comment:
Seems we're missing a `impl From<zbus_xml::Error> for zbus::Error`, which would make this `map_err` call redundant. Could you please add that instead (in a separate commit that comes before this one)?

Reply:
Not sure where to add it, zbus_xml does not use zbus, and zbus only use zbus_xml for testing.. Seems a feature gate might be needed for either case.

Reply:
> zbus only use zbus_xml for testing

Right i forgot zbus doesn't depend on zbus_xml.